packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -277,8 +277,11 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
       'width' | 'colSpan' | 'showRightBorder' | 'indexRelativeToAllColumns'
     >,
   ) => {
+    const isReorderCell = column.field === '__reorder__';
```

Comment:
This whole condition is way to complex and hard to read.
IMHO it should be a full method or at list a variable assignment.

Reply:
thats true ... I'll check if I can extract that into a helper

Reply:
OK, I did look around a bit.
https://github.com/mui/mui-x/blob/891cea1809e44f71a270dc3f512528b5db7717d7/packages/grid/x-data-grid/src/components/GridRow.tsx#L280-L285
Is basically duplicated in the drag handlers. That seems a bit weird. 
My proposal would be to allow the drag events to being published, but handle the event execution like before.
For reference the row reorder handles it here:
https://github.com/mui/mui-x/blob/891cea1809e44f71a270dc3f512528b5db7717d7/packages/grid/x-data-grid-pro/src/hooks/features/rowReorder/useGridRowReorder.tsx#L75-L77

And the column reorder here:
https://github.com/mui/mui-x/blob/891cea1809e44f71a270dc3f512528b5db7717d7/packages/grid/x-data-grid-pro/src/hooks/features/columnReorder/useGridColumnReorder.tsx#L88-L90

I would still keep the `disableDragEvents` variable for this case:
```
const disableDragEvents = Object.keys(editRowsState).length > 0;
```

@flaviendelangle WDYT?

Reply:
Adding some context on why we disable drag handlers: https://github.com/mui/mui-x/issues/4841

Reply:
Since disabling the dragEvents is beneficial (thanks for pointing that out @cherniavskii) I just renamed and restructured it a tiny bit to make it more readable at least.